### PR TITLE
Allow NPM packages to build for production

### PR DESCRIPTION
**Pull request by joelpittet**
# Problem/Motivation

We were noticing a bunch of security issues through `npm audit` and couldn't get `npm run production` to work. I'll keep this patch as small as possible to get the `npm run production` working again.  Will create follow-ups with the security stuff after it's working again.

## Matching Lavavel 6.x changes
https://github.com/laravel/laravel/blob/6.x/package.json

NPM script changed slightly to remove the `--hide-modules`

Laravel Mix has more fixes in the `6.x` branch but larger change, so tried to go with least changes to get this accepted 🤞. Matched with `5.x`

## Packages 
`css-loader` excplicitly added because 5.0.2 is throwing `TypeError: text.forEach is not a function` errors on the less files.

Bootstrap-table removed some CSS and fixed a typo in their path: `extentions` fixed to `extensions`

## CSS files

Signature pad CSS was missing and the minified file was empty, it was copying to `build` instead of `dist`, so I changed it and the blade file referencing it.

## JavaScript files
Resources path were incorrect missing `assets` directory for FileSaver.min.js

`  './resources/assets/js/FileSaver.min.js',`

resources/assets/js/xlsx.core.min.js missing so added the library instead.
`npm i sheetjs`

## Less
Less and `less-loader`gave me the most grief ...
Moved LESS from hash to latest 4.x stable release (from approx 2.x)
https://github.com/less/less.js/commit/efa6eb5306f28a7ef7e235d79ce854b780345591

## Todo
I left some changes out so this is easier to test and get through, probably need to make some follow-ups after this is in.

- `admin-lte` has updates with security fixes but didn't test the changes, most of the security is due to fullcalendar
- Worth considering adding `signature_pad` as a dependency, but upgrade testing needed `npm install --save signature_pad`
- I left `phantomjs` but it's been deprecated for a number of years, maybe we can move to chromedriver or something else? There are some dependencies there that have security
- `papaparse` Doesn't seem to be used at all and has a bunch of vulnerabilities, remove?
- `babel-preset-latest` is deprecated and should be in the main babel now. Remove?

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

It has only been tested in that `npm run production` now works, and the same files are created via a diff of

`git diff --color-words public/mix-manifest.json`

### CODE REVIEW

**Comment by joelpittet** [comment]
Ok sorry, my PR in #9169 is wrong and my force-push killed it...

**Comment by joelpittet** [comment]
My first commit doesn't have any asset compilation commits to keep it easy to review, but LMK if you want the results of `npm run production` committed as well.  I will commit 
`public/css/dist/signature-pad.min.css` because it's new, and `public/mix-manifest.json` because it's changed with that file too.

This unfortunately will conflict with my other simple PR #9170 because the same line is touched.

**Comment by joelpittet** [comment]
I added updates to `papaparse` and `axios` to the commit because they seemed fairly harmless to bump and had security vulnerabilities attached in dependencies.  

Most of the rest are related to `admin-lte` and `jquery-form-validator` and `phantomjs` but they probably need more testing.

**Comment by joelpittet** [comment]
Actually scratch that, the scope of this PR is to get `npm run production` working again. I removed the security fixes and will do those in a follow-up.

**Comment by joelpittet** [comment]
@snipe I just did a visual comparison of the JS/CSS output before and after... the output looks good, mostly removing old prefixes in CSS. I was able to get `npm run production` so the point behind this PR is now moot... hope the work is not in vein it's dependency hell trying to update laravel-mix and company.  I'll need to take direction from you or @uberbrady
on which way you'd like to take this.

File: public/mix-manifest.json [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+    "/css/dist/skins/skin-blue-dark.min.css": "/css/dist/skins/skin-blue-dark.min.css?id=8ef6d8a72b7f9719ea0e",
+    "/css/dist/skins/skin-yellow-dark.min.css": "/css/dist/skins/skin-yellow-dark.min.css?id=da122cc1c3e585e888f5",
+    "/css/dist/skins/skin-red-dark.min.css": "/css/dist/skins/skin-red-dark.min.css?id=b8776482134bff15a82b",
+    "/css/dist/skins/skin-purple-dark.min.css": "/css/dist/skins/skin-purple-dark.min.css?id=e80057f65c17cff032e0",
+    "/css/dist/skins/skin-orange-dark.min.css": "/css/dist/skins/skin-orange-dark.min.css?id=90bdd43a4b1daa42120f",
+    "/css/dist/skins/skin-contrast.min.css": "/css/dist/skins/skin-contrast.min.css?id=9fa87f2673f15e56aed6",
     "/css/dist/signature-pad.css": "/css/dist/signature-pad.css?id=6a89d3cd901305e66ced",
-    "/css/build/signature-pad.min.css": "/css/build/signature-pad.min.css?id=d41d8cd98f00b204e980",
-    "/css/dist/bootstrap-table.css": "/css/dist/bootstrap-table.css?id=1e77fde04b3f42432581",
+    "/css/dist/signature-pad.min.css": "/css/dist/signature-pad.min.css?id=6a89d3cd901305e66ced",
```
Comment (joelpittet) [inline public/mix-manifest.json:33]: Changed to dist like the rest of them.

**Comment by joelpittet** [comment]
Here's a nice breakdown of the impact for security and bringing things up-to-date (a bit more)
snipe-it_and_snipe-it_and_hardware-inventory

**Comment by joelpittet** [comment]
Hope I'm going in the right direction on this, @uberbrady  thoughts?

**Comment by snipe** [comment]
You are such a peach, thank you so much! Sorry to have this take so long. We’ve been on-boarding a new person and trying to handle some pre-v5 PRs that are a bit stale, and couldn’t come up for air right away. We’ll be addressing this PR this week, without question. This *looks* good to me, but we’ve got to finish our smoke tests to make sure none of these updates break in weird, unexpected ways. Because JavaScript, that’s why. 

Thanks again for your ongoing patience and for helping out with the project <3

**Comment by snipe** [comment]
Nice work!

**Comment by joelpittet** [comment]
Thanks @snipe! I'll see if I can help the sec issues in the admin theme next (in their repo).

**Pull request merged**